server/migrations/201901290642-init.js

```diff
@@ -0,0 +1,164 @@
+                },
+                type: {
+                    type: Sequelize.STRING,
+                    allowNull: false,
+                },
+                token: {
+                    type: Sequelize.STRING,
+                    allowNull: false,
+                },
+                isArchived: {
+                    type: Sequelize.BOOLEAN,
+                    defaultValue: false,
+                    allowNull: false,
+                },
+                createdAt: {
+                    type: Sequelize.DATE,
+                    allowNull: false,
+                },
+                updatedAt: {
+                    type: Sequelize.DATE,
+                    allowNull: false,
+                },
+                UserId: {
+                    type: Sequelize.INTEGER,
+                    onUpdate: 'CASCADE',
+                    onDelete: 'SET NULL',
+                    references: {
+                        model: 'Users',
+                        key: 'id',
+                    },
+                    allowNull: true,
+                },
+            },
+            {},
+        ],
+    },
+    {
+        fn: 'createTable',
+        params: [
+            'Notifications',
+            {
+                id: {
+                    type: Sequelize.INTEGER,
+                    primaryKey: true,
+                    autoIncrement: true,
+                },
+                type: {
+                    type: Sequelize.STRING,
+                    allowNull: false,
+                },
+                payload: {
+                    type: Sequelize.JSON,
+                    allowNull: false,
+                },
+                status: {
+                    type: Sequelize.STRING,
+                    allowNull: false,
+                },
+                createdAt: {
+                    type: Sequelize.DATE,
+                    allowNull: false,
+                },
+                updatedAt: {
+                    type: Sequelize.DATE,
+                    allowNull: false,
+                },
+                DeviceId: {
+                    type: Sequelize.INTEGER,
+                    onUpdate: 'CASCADE',
+                    onDelete: 'SET NULL',
+                    references: {
+                        model: 'Devices',
+                        key: 'id',
+                    },
+                    allowNull: true,
+                },
+            },
+            {},
+        ],
+    },
+    {
+        // This manual query is required, rather than using sequelize's addIndex()
+        // because addIndex() does not support IF NOT EXISTS.
+        fn: 'query',
+        query: 'CREATE INDEX IF NOT EXISTS users_uuid ON public."Users" USING btree (uuid);',
+    },
+];
+
+module.exports = {
+    pos: 0,
+    up(queryInterface) {
+        let index = this.pos;
+        return new Promise((resolve, reject) => {
+            function next() {
+                if (index < migrationCommands.length) {
```

Comment:
Why not iterate on the array directly with something like `forEach`?

Reply:
Since production (the Dockerfile) uses Node 8 and since most of our devs are already using Node 8 (and should be if they are not already), I'm going to use `async/await` in the migrations to make them read much more cleanly.